File: Editorial/0100_Same_Tree/Go/1_Recursion.go
func isSameTree(p *TreeNode, q *TreeNode) bool {
    // p and q are both null
    if p == nil && q == nil {
        return true
    }
    // one of p and q is null
    if (q == nil) || (p == nil) {
        return false
    }
    if p.Val != q.Val {
        return false
    }
    return isSameTree(p.Right, q.Right) &&
        isSameTree(p.Left, q.Left)
}

File: Editorial/0100_Same_Tree/C/1_Recursion.c
bool isSameTree(struct TreeNode* p, struct TreeNode* q) {
    // p and q are both null
    if (p == NULL && q == NULL) return true;
    // one of p and q is null
    if (q == NULL || p == NULL) return false;
    if (p->val != q->val) return false;
    return isSameTree(p->right, q->right) && isSameTree(p->left, q->left);
}
